internal/verifier/change_stream_test.go

```diff
@@ -291,3 +307,36 @@ func (suite *IntegrationTestSuite) TestEventBeforeWritesOff() {
 
 	suite.Assert().Equal(docsCount, totalFailed, "all source docs should be missing")
 }
+
+func (suite *IntegrationTestSuite) TestCreateForbidden() {
+	ctx := suite.Context()
+	buildInfo, err := util.GetBuildInfo(ctx, suite.srcMongoClient)
+	suite.Require().NoError(err)
+
+	if buildInfo.VersionArray[0] < 6 {
+		suite.T().Skipf("This test requires server v6+. (Found: %v)", buildInfo.VersionArray)
+	}
+
+	verifier := suite.BuildVerifier()
+
+	// start verifier
+	verifierRunner := RunVerifierCheck(suite.Context(), suite.T(), verifier)
+
+	// wait for generation 0 to end
+	verifierRunner.AwaitGenerationEnd()
+
+	db := suite.srcMongoClient.Database(suite.DBNameForTest())
+	coll := db.Collection("mycoll")
+	suite.Require().NoError(
+		db.CreateCollection(ctx, coll.Name()),
+	)
+
+	suite.Require().NoError(verifier.WritesOff(ctx))
+
+	err = verifierRunner.Await()
```

Comment:
I don't think a change stream error will be returned until writesOff

Reply:
It’s actually indeterminate, it seems: the error may happen before writesOff, or it may happen when shutting down the change stream.